networks/2b_community_detection.py

```diff
@@ -0,0 +1,202 @@
+# -*- coding: utf-8 -*-
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:core_acc] *
+#     language: python
+#     name: conda-env-core_acc-py
+# ---
+
+# # Network community detection
+#
+# This notebook performs community detection approaches to identify network modules.
```

Comment:
What is basically the difference between applying clustering and community detection algorithms? I never used the latter, and I wonder what is the main difference.

Reply:
Good question! This was my same question as well when Casey mentioned I might want to take a look at community detection methods. So the main difference seems to be that clustering thinks about each gene as a point/entity and performs clustering to place similar genes (i.e. genes that are close together in correlation space) together where each gene belongs to a single module. Whereas, community detection considers genes to be in a network/graph where genes are connected to other genes based on similarity between expression pattern across samples (i.e. correlation score between gene A and B). Community detection will define modules as a group of genes that are densely connected to each other but sparsely connected to other genes in the network (within vs between edges). 

I can update this description to include this information